### ISSUE TITLE
Error in sampling the data

### ISSUE THREAD

**Issue by parvaneh-soleimany** [tabularis-ai/be_great#55]
Hey. I am facing error on sampling data. 
I get error: "Breaking the generation loop!"
and when I checked the code I understood in the next line, df_gen gets empty and it is obviously because the model just generated placeholder, and not the desired data.
df_gen = df_gen[~(df_gen == "placeholder").any(axis=1)]

I am working on it for more than one week but I couldn't handle it.
Can you please help me? I am really in need of this code to works.
My data is a tabular text data, each column consisting number or piece of text (sometimes long texts).

Thank you in advance.

**Comment by parvaneh-soleimany** [comment]
No one has the same issue?

**Comment by unnir** [comment]
>I got the same error. The only pattern I could identify so far is that it only happens with datasets having >=20 features.

This is correct, some feature names can exceed the context windows of an LLM model, thus, you want be able to generate synthetic samples. 

Another solution is to use new (and better) models for example:

```py
from be_great import GReaT
from sklearn.datasets import fetch_california_housing

data = fetch_california_housing(as_frame=True).frame

model = GReaT(llm='unsloth/Llama-3.2-1B', batch_size=32,  epochs=1, fp16=True, report_to="none")
model.fit(data)
synthetic_data = model.sample(n_samples=100) 
```
